src/dataIntegration/transcriptomics/call_GIMME.m

```diff
@@ -0,0 +1,286 @@
+
+    %first make model irreversible
+    [modelIrrev,matchRev,rev2irrev,irrev2rev] = convertToIrreversible(model); %% CHECK with the last version of this function in cobra v3 to see if we can remove the modified version included in this script
+
+    nbExpressionRxns = size(expressionRxns,1);
+    if (nbExpressionRxns < nRxns)
+        display('Warning: Fewer expression data inputs than reactions');
+        expressionRxns(nbExpressionRxns+1:nRxns,:) = zeros(nRxns-nbExpressionRxns, size(expressionRxns,2));
+    end
+
+    nIrrevRxns = size(irrev2rev,1);
+    expressionRxnsIrrev = zeros(nIrrevRxns,1);
+    for i=1:nIrrevRxns
+        expressionRxnsIrrev(i,1) = expressionRxns(irrev2rev(i,1),1);
+    end
+
+    nObjectives = size(objectiveCol,1);
+    for i=1:nObjectives
+        objectiveColIrrev(i,:) = [rev2irrev{objectiveCol(i,1),1}(1,1) objectiveCol(i,2)];
+    end
+
+    %Solve initially to get max for each objective
+    for i=1:size(objectiveCol)
+        %define parameters for initial solution
+        modelIrrev.c=zeros(nIrrevRxns,1);
+        modelIrrev.c(objectiveColIrrev(i,1),1)=1;
+
+        %find max objective
+        FBAsolution = optimizeCbModel(modelIrrev);
+        if (FBAsolution.stat ~= 1)
+            not_solved=1;
+            display('Failed to solve initial FBA problem');
+            return
+        end
+        maxObjective(i)=FBAsolution.f;
+    end
+
+    model2gimme = modelIrrev;
+    model2gimme.c = zeros(nIrrevRxns,1);
+
+
+    for i=1:nIrrevRxns
+        if (expressionRxnsIrrev(i,1) > -1)   %if not absent reaction
+            if (expressionRxnsIrrev(i,1) < cutoff)
+                model2gimme.c(i,1) = cutoff-expressionRxnsIrrev(i,1); %FIX: use expression level as weight
+            end
+        end
+    end
+
+    for i=1:size(objectiveColIrrev,1)
+        model2gimme.lb(objectiveColIrrev(i,1),1) = objectiveColIrrev(i,2) * maxObjective(i);
+    end
+
+    gimmeSolution = optimizeCbModel(model2gimme,'min');
+
+    if (gimmeSolution.stat ~= 1)
+    %No solution for the problem
+        display('Failed to solve GIMME problem'); 
+        gimmeSolution.x = zeros(nIrrevRxns,1);
+    end
+
+    reactionActivityIrrev = zeros(nIrrevRxns,1);
+    for i=1:nIrrevRxns
+        if ((expressionRxnsIrrev(i,1) > cutoff) | (expressionRxnsIrrev(i,1) == -1))
+            reactionActivityIrrev(i,1)=1;
+        elseif (gimmeSolution.x(i,1) > 0)
+            reactionActivityIrrev(i,1)=2;
+        end
+    end
+
+    %Translate reactionActivity to reversible model
+    reactionActivity = zeros(nRxns,1);
+    for i=1:nRxns
+        for j=1:size(rev2irrev{i,1},2)
+            if (reactionActivityIrrev(rev2irrev{i,1}(1,j)) > reactionActivity(i,1))
+                reactionActivity(i,1) = reactionActivityIrrev(rev2irrev{i,1}(1,j));
+            end
+        end
+    end
+
+end
+
+function [modelIrrev,matchRev,rev2irrev,irrev2rev] = convertToIrreversible(model)
+%convertToIrreversible Convert model to irreversible format

```

Comment:
yes, i will provide some fix to convertToIrreversible. Just wanted to keep that before the changes are merged to not forget it later ;)